data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/SelectEntriesProcessor.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.mutateevent;
+
+import org.opensearch.dataprepper.expression.ExpressionEvaluator;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.processor.AbstractProcessor;
+import org.opensearch.dataprepper.model.processor.Processor;
+import org.opensearch.dataprepper.model.record.Record;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+@DataPrepperPlugin(name = "select_entries", pluginType = Processor.class, pluginConfigurationType = SelectEntriesProcessorConfig.class)
+public class SelectEntriesProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
+    private final List<String> entries;
+    private final String selectWhen;
+
+    private final ExpressionEvaluator expressionEvaluator;
+
+    @DataPrepperPluginConstructor
+    public SelectEntriesProcessor(final PluginMetrics pluginMetrics, final SelectEntriesProcessorConfig config, final ExpressionEvaluator expressionEvaluator) {
+        super(pluginMetrics);
+        this.entries = Arrays.asList(config.getIncludeKeys());
+        this.selectWhen = config.getSelectWhen();
+        this.expressionEvaluator = expressionEvaluator;
+    }
+
+    @Override
+    public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
+        for(final Record<Event> record : records) {
+            final Event recordEvent = record.getData();
+
+            if (Objects.nonNull(selectWhen) && !expressionEvaluator.evaluateConditional(selectWhen, recordEvent)) {
+                continue;
+            }
+            // To handle nested case, just get the values and store
+            // in a temporary map.
+            Map<String, Object> outMap = new HashMap<>();
+            for (String entry: entries) {
+                Object val = recordEvent.get(entry, Object.class);
+                if (val != null) {
+                    outMap.put(entry, val);
+                }
+            }
+            // Delete all entries from the event
+            Set keysToDelete = recordEvent.toMap().keySet();
+            Iterator iter = keysToDelete.iterator();
+            while (iter.hasNext()) {
```

Comment:
It might make more sense to have a `clear()` method. We could add that.

Reply:
I thought about. OK, will add taht to JacksonEvent.

But is this is the optimal way or just creating a new event with the entries of interest is better>

Reply:
I'm not sure which way is optimal from a performance perspective. But, in terms of tracking all the metadata and keeping the handles, wouldn't clearing and setting be more straightforward?